src/app/app.authorization.js

```diff
@@ -0,0 +1,13 @@
+angular.module('app').run([
+  '$rootScope', '$state', 'authenticationService', '$http', 'localStorageService',
+  function ($rootScope, $state, authenticationService) {
+
+  $rootScope.$on('$stateChangeStart',
+    function (event, toState, toParams, fromState, fromParams) {
+      if (!authenticationService.isLoggedIn() && toState.data.requireLogin) {
```

Comment:
What about adding another flag like 'requireLogin' that will be used for these kind of states? The ones that can be accessed by either logged or not logged users.